File: src/main/java/io/github/flegacy/flms/ui/element/TransferButton.java
package io.github.flegacy.flms.ui.element;

import org.bukkit.entity.Player;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.inventory.ItemStack;

import io.github.flegacy.flms.ui.FLMSInterface;
import io.github.flegacy.flms.utils.SoundPlayer;

public class TransferButton implements ClickableElement {

    private final ItemStack display;
    private final FLMSInterface next;

    public TransferButton(ItemStack display, FLMSInterface next) {
        this.display = display;
        this.next = next;
    }

	@Override
	public ItemStack displayItem() {
        return display.clone();
	}

	@Override
	public void execute(InventoryClickEvent event) {
        if (!(event.getWhoClicked() instanceof Player player))
            return;

        SoundPlayer.playButtonClickSound(player);
        next.open(player);
	}

}

File: src/main/kotlin/io/github/flegacy/flms/ui/element/TransferButton.kt
package io.github.flegacy.flms.ui.element

import io.github.flegacy.flms.ui.FLMSInterface
import io.github.flegacy.flms.util.soundClick
import org.bukkit.entity.Player
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.inventory.ItemStack

class TransferButton(private val display: ItemStack, private val goto: FLMSInterface): ClickableElement {

    override fun getDisplay(): ItemStack {
        return display.clone()
    }

    override fun onClick(event: InventoryClickEvent) {
        val player = event.whoClicked as? Player ?: return
        soundClick(player)
        goto.open(player)
    }

}
